defaults/main.yml

```diff
@@ -23,30 +23,53 @@ datadog_agent_binary_path_windows: "C:\\Program Files\\Datadog\\Datadog Agent\\b
 # list of additional groups for datadog_user
 datadog_additional_groups: {}
 
-# default apt repo
-datadog_apt_repo: "deb https://apt.datadoghq.com/ stable 6"
+# Major version of the Agent that will be installed.
+# Possible values: 5, 6, 7
+datadog_agent_major_version: 7
+
+# Pin agent to a version. Highly recommended.
+# Defaults to the latest version of the major version chosen in datadog_agent_major_version
+# The version pinned needs to have the same major as datadog_agent_major_version
```

Comment:
I thought it was one or the other from memory of the RFC. It doesn't make sens for ansible to duplicate both and it's error prone: what if I set major to 6 and pin 7 with the version.

I'm sure it make sens for chef, but with ansible it complexify everything for no gain.

Reply:
Take the new `agent_major_version` as the previous `datadog_agent5` variable we already had, just that now it can have three values instead of two so it can't be a boolean anymore. We aren't adding more complexity, only changing a type. If you use the default (agent 7) you don't even need to specify it, like it was the case before.

I can see the duplicity in specifying the version, but it's not worse than the previous boolean we had. No approach is perfect and there are pros and cons to both, but we discussed this a while back and got to this conclusion.

For example, having a single variable doesn't let you pin "latest agent 6", which would be a regression (unless we allowed to use `6.x` or something like that, but that adds a lot more complexity). This change also keeps the code changes to the minimum, since it's only changing the boolean check to an integer check.

Also, keeping Chef, Puppet and Ansible similar is also a benefit IMO.

Reply:
I agree it's not worth than before but since we're going to bump the major we have an opportunity to make is way better ! It would not be a lot of work to deduce the major from the pinned version (literally one ansible task) and would greatly improve our user experience. Also this would be the perfect timing to abstract the difference in the version format between each OS (`6.15.0-1`, `1:6.15.0-1`, `6.15.0` ...).

And I don't think we should have Chef way of doing things in Ansible for example. We should be aiming at a native way of doing things. Users don't use chef and ansible and puppet at the same time, they only use one of those. Aligning behavior between all of those tools is a improvement for us but a drawback for our user.

In the end I think having `agent_major_version` and `agent_version` is a good idea (hence my validating the RFC), I just think we should deduce one when the other is pinned. Respecting the RFC but also making it intuitive to use.

Reply:
What do we do if the user specifies both, then? Which one should take precedence? Isn't that more unpredictable from a user perspective?

I don't think we are doing things "the Chef way" on Ansible: this is not the Chef way either. This is a tool-agnostic solution that we decided to use across all config management tools and I have a strong opinion on not having Ansible be different from the others.

Unifying version format (`6.15.0`, `1:6.15.0-1`...) is something that we could have done but didn't make it into the proposal, so I also think it is late to propose it. The change should be backwards compatible if possible and consistent across all config management tools (that's why we made a unified proposal) and not something to add for Ansible only in a code review.

Reply:
ok, so I see 2 things being discussed here:

* inferring `agent_major_version` from `agent_version`: we can do this later if we think it's worth it, it would not be a breaking change anyway, so I don't think we should try to implement this now.
* abstracting out the difference in the version format between OSes: we could consider doing this indeed, it could make life easier for users. I'd find a way to make this change in a non-breaking way though, to make the upgrade path easier. Maybe something for a future minor version.

Reply:
Regarding backward compatibility we're going to break the behavior anyway, by installing 7 by default (and therefor bump the major). This mean a manual update of the role for user. We won't be able to break backward compatibility again later (or we would have to bump the major again). I think now is a good time to implement such simple changes.

Regarding:
> What do we do if the user specifies both, then? Which one should take precedence? Isn't 
> that more unpredictable from a user perspective?

Simple: we raise an error or ignore `major` (I vote for raising an error). They either use `major` to pin latest or use `version` to pin a specific version.

Reply:
As discussed, this will be implemented in a separate PR.